### ISSUE TITLE
<inputNumber> format is not respected

### ISSUE THREAD

**Issue by schlebe** [primefaces/primefaces#2537]
Hello,

I have set the "text" value of an input number to "12.00" and the "decimalPlaces" attribute is equal to "2".

After pressing a SUBMIT Button, I don't have "12.00" but "12" !

The <inputNumber> element seems converting my string to a number and forgetting to respect my number format that is with 2 decimals.

I have no problem when number value is "12.34"

## 1) Environment
- PrimeFaces version: 6.1.1
- Affected browsers: Opera/Chrome

## 2) Expected behavior

I expect that <inputNumber> element respects my formatting.

My formatting is "0,##", if the displayed value is 12.00, than the text value returned after SUBMIT must be "12.00" and not "12" because "text" type is String and not Double.

## 3) Actual behavior

The input number returned by the oController.getText() method is a number with 2 decimal. The value displayed on the browser is a number with 2 decimal BUT the number returned after pressing a SUBMIT button is a number WITHOUT decimal.

## 4) Steps to reproduce

Simply verify the String value linked to the widget in bean after pressing a SUBMIT button.

## 5) Sample XHTML

```
<p:inputNumber
    id="txtRisk" 
    value="#{oPostLabActionController.panelTpl.txtRisk.text}"
    rendered="#{!oPostLabActionController.panelTpl.txtRisk.protected}"
    minValue="0.00"
    maxValue="100.00"
    emptyValue="empty"
    decimalPlaces="2"
    />
```
## 6) Sample bean

```
   public void setText(String s)
        {
        sText = s;
        }
    
    public String getText() 
        { 
        return sText;
        }
```

**Comment by tandraschko** [comment]
Thats not in our hand.
The jquery autonumeric plugin cuts of the zeros on the right side when writing it back to the hidden input.
Also it's quite unnormal to work with a String as variable.

I would use a converter if you need to re-add zeros on the right side.

**Comment by schlebe** [comment]
Thanks for your help but I have already a workaround for my problem.
In fact, I compare in bean the String passed to widget before displaying form with the string returned after submitting the form. 
If the 2 string are not equal then the value/text contained in widget has been changed.
I have only problem with <textEditor> and &lt;inputNumber&gt; that change returned value after posting the form.

QUESTION: who is responsible of &lt;inputNumber&gt; widget ?

**Comment by tandraschko** [comment]
the autonumeric jquery plugin is resposible.
But IMO this is a won't fix as 12 and 12.00 is the same number. And the component is created for handling numbers ;)

**Comment by schlebe** [comment]
ok thanks for the information.
You have right ... the value of inputNumber in Primefaces is a Double and not a String.

On JQuery, the get() method return a String but the "value" attribute in Primefaces on a lot of widgets is an objet or an internal type (Integer, Boolean, String) that is not always a String like I thought incorrectly.
